### PR TITLE
exp(stage4): roberta-base amharic baseline results (#32)

### LINKED ISSUE #32: [EXP] Train Stage 4 baseline: RoBERTa-base Amharic sentiment classification
**Stage:** stage:4-sentiment

**Hypothesis**
Monolingual \
asyosef/roberta-base-amharic\ (110M) with Amharic-native subword tokenization provides superior representation quality for 3-class sentiment classification over explicit aspect-opinion pairs compared to multilingual AfroXLMR-base (278M) and lightweight bert-small-amharic (28M).

**Config used**
\configs/stage4_roberta_amharic.yaml\

**Dataset split**
\data/prepared/{train,test}.jsonl\

**Objective**
Train and record evaluation metrics (accuracy, macro F1, per-class breakdown with focus on NEUTRAL_f1) and checkpoints under \
esults/stage4/roberta_base_run1/\.

### PR DESCRIPTION
**Pull request by AzariyasMekonen**
Closes #32.

### Experiment Results: Stage 4 Sentiment Classification (RoBERTa-base Amharic)

- **Model**: \
rasyosef/roberta-base-amharic\
- **Config**: \configs/stage4_roberta_amharic.yaml\
- **Test Set Accuracy**: **89.00%**
- **Macro F1**: **0.7103**

#### Per-Class Breakdown
| Sentiment | Precision | Recall | F1 | Support |
| :--- | :--- | :--- | :--- | :--- |
| **NEGATIVE** | 93.17% | 93.83% | **0.9350** | 1,992 |
| **POSITIVE** | 88.31% | 83.48% | **0.8583** | 896 |
| **NEUTRAL** | 29.23% | 40.00% | **0.3378** | 95 |

Checkpoint artifacts saved under \
esults/stage4/roberta_base_run1/\.

## Summary by CodeRabbit

- **Documentation**
  - Added recorded evaluation metrics for negative, neutral, and positive sentiment classes, including precision, recall, F1, support, macro-F1, and accuracy.
  - Added the resolved configuration for the stage-4 RoBERTa training run, including model, data paths, training settings, and reproducibility parameters.
  - Added tokenizer configuration covering special tokens, sequence length limits, backend behavior, and cleanup settings.

### CODE REVIEW

File: results/stage4/roberta_base_run1/resolved_args.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "config": null,
+  "train": "data/prepared/train.jsonl",
+  "test": "data/prepared/test.jsonl",
+  "model_name": "rasyosef/roberta-base-amharic",
```
[the remark this answers is not in the document]
Comment (AzariyasMekonen) [reply results/stage4/roberta_base_run1/resolved_args.json:5]: changed the pr objective to rasyosef/roberta-base-amharic
